ndindex/ndindex.py

```diff
@@ -65,6 +66,91 @@ def __init__(self, f):
     def __get__(self, obj, owner):
         return self.f(owner)
 
+class _Guard:
+    def __init__(self):
+        self.val = False
+
+    def __call__(self):
+        if self.val:
+            return True
+        else:
+            self.val = True
+            return False
+
+def parse_index(node_or_string):
+    """
+    "Safely" (needs validation) evaluate an expression node or a string containing
+    a (limited subset) of valid numpy index or slice expressions.
```

Comment:
The docstring should probably clarify what is and isn't allowed. I guess we aren't allowing any kind of expressions that create numbers. 

Actually, we may at one point want to support just anything that parses, and make the current behavior under a `safe=True` flag, which I guess should be the default. But we can worry about that later. 

Also this docstring needs examples. and should be added to Sphinx.